### PR TITLE
journalctl: filtering by facility

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by joshtriplett** [comment]
Very nice! This will help people who are used to facility-specific logfiles in syslog.

**Comment by boucman** [comment]
Can I ask naively what is the added value compared to filtering by SYSLOG_FACILITY ? is the main point to make it easier to find because it's in the man pages (which would be a valid reason)

**Comment by dnicolodi** [comment]
@boucman It adds the capability of resolving facility names to facility numbers.

**Comment by keszybz** [comment]
> @boucman It adds the capability of resolving facility names to facility numbers.

That is the main reason. It's also easier to filter by mutliple facilities: `--facility daemon,mail` is arguably nicer than `SYSLOG_FACILITY=2 SYSLOG_FACILITY=12`.

File: man/journalctl.xml [hunk 1]
```diff
@@ -597,6 +597,16 @@
         priorities.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--facility=</option></term>
+
+        <listitem><para>Filter output by syslog facility. Takes a comma-separated list of numbers of facility
```
Comment (yuwata) [inline man/journalctl.xml]: Maybe, `numbers _or_ facility names.`

File: src/journal/journalctl.c [hunk 2]
```diff
@@ -1676,6 +1731,26 @@ static int add_priorities(sd_journal *j) {
...
+                char match[STRLEN("SYSLOG_FACILITY=") + DECIMAL_STR_MAX(int)];
+
+                xsprintf(match, "SYSLOG_FACILITY=%d", PTR_TO_INT(p));
+
+                r = sd_journal_add_match(j, match, strlen(match));
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add match: %m");
+        }
+
+        arg_facilities = set_free(arg_facilities);
```
Comment (yuwata) [inline src/journal/journalctl.c]: This should be moved to after `finish:` in `main()`. Otherwise, e.g., `--facility=mail,help` causes memleak.

**Comment by keszybz** [comment]
Thanks, updates with the two requested fixes.

**Pull request merged**